x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -300,6 +301,7 @@ public void createApiKey(
     ) {
         assert request.getType() != ApiKey.Type.CROSS_CLUSTER || userRoleDescriptors.isEmpty()
             : "owner user role descriptor must be empty for cross-cluster API keys";
+        assert false == hasWorkflowsRestriction(userRoleDescriptors) : "workflows restriction are not supported by user role descriptors";
```

Comment:
User roles can be resolved by custom role providers. Do you plan to prevent them returning workflow roles? Otherwise, we may need more than an assertion here.

Reply:
Good question. I think we should prevent it. I will convert this into `IllegalArgumentException`.